sqlmesh/core/table_diff.py

```diff
@@ -317,18 +317,24 @@ def name(e: exp.Expression) -> str:
                     *(exp.column(c) for c in source_schema),
                     self.source_key_expression.as_(SQLMESH_JOIN_KEY_COL),
                 )
-                .from_(self.source_table)
+                .from_(self.source_table.as_("s"))
                 .where(self.where)
             )
             target_query = (
                 exp.select(
                     *(exp.column(c) for c in target_schema),
                     self.target_key_expression.as_(SQLMESH_JOIN_KEY_COL),
                 )
-                .from_(self.target_table)
+                .from_(self.target_table.as_("t"))
                 .where(self.where)
             )
 
+            # Ensure every column is qualified with the alias in the source and target queries
+            for col in source_query.find_all(exp.Column):
+                col.set("table", exp.to_identifier("s"))
```

Comment:
can you do this in the builder instead in the exp.column statement?

Reply:
I originally did that but realized it's not as straightforward because we also [include a more complex expression](https://github.com/TobikoData/sqlmesh/blob/43655b285994e43ee23447f9c46adffeac491564/sqlmesh/core/table_diff.py#L318) that's [constructed elsewhere](https://github.com/TobikoData/sqlmesh/blob/43655b285994e43ee23447f9c46adffeac491564/sqlmesh/core/table_diff.py#L227-L246).

Reply:
Update– it seems like we don't need to worry about the source/target key expressions, because these are properly aliased and so shouldn't clash with any of these tables' columns. I'll apply Toby's suggestion.